pkg/build/controller/scheduler/scheduler.go

```diff
@@ -0,0 +1,143 @@
+package scheduler
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/golang/glog"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client"
+)
+
+type Scheduler interface {
+	Next(*buildapi.Build) (bool, error)
+}
+
+type ParallelScheduler struct{}
+
+func (s *ParallelScheduler) Next(build *buildapi.Build) (bool, error) {
+	if buildutil.BuildSchedulePolicy(build) != buildapi.BuildSchedulePolicyParallel {
+		return false, nil
+	}
+	glog.V(5).Infof("Build %s/%s scheduler is parallel", build.Namespace, build.Name)
+	return true, nil
```

Comment:
> so you won't be able to schedule any parallel build till you have running serial build?  

?

you mean "you won't be able to run any parallel builds while you have a running serial build?"  If so then yes, that's what i mean.

> when I change my BC to parallel, I expect new builds to run immediately in parallel with whatever is running currently :-)

when i change my BC to parallel, i expect builds i ran as serial builds to continue to run in isolation.